src/pages/index.js

```diff
@@ -0,0 +1,346 @@
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalcloseBtn = previewModal.querySelector(
+  ".modal__close-btn_preview"
+);
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+function handleLike(evt, id, cardLikedBtn){
+  const isLiked = cardLikedBtn.classList.contains('card__like-button_liked');
+  
+  api. handleLikeStatus(id, isLiked) 
+  .then(() => {
+    cardLikedBtn.classList.toggle("card__like-button_liked");
+  })
+  .catch(console.error);
+}
+
+previewModalcloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+let selectedCard, selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  
+  if (data.isLiked) { 
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikedBtn.addEventListener("click", (evt) => 
+    handleLike(evt, data._id, cardLikedBtn));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () => 
+    handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+
+  return cardElement;
+}
+
+ 
+
+function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+  
+  setButtonText(cardDeleteBtn, true, "Delete", "Deleting...");
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  if (avatarDeletetBtn) {
+    setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+  }
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      
+      closeModal(deleteModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(avatarDeletetBtn, false, "Delete", "Deleting...");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  //change text content to "Saving...""
+  const cardSubmitBtn = evt.submitter; 
```

Comment:
`cardSubmitBtn `


This is not about the card. It's about the `profile`


`profileSubmitBtn`